gradio/components/file_explorer.py

```diff
@@ -85,7 +86,7 @@ def __init__(
             )
             root_dir = root
             self._constructor_args[0]["root_dir"] = root
-        self.root_dir = os.path.abspath(root_dir)
+        self.root_dir = cast(DeveloperPath, os.path.abspath(root_dir))
```

Comment:
could we do `DeveloperPath(os.path.abspath(root_dir))` instead of using casting?

Reply:
Also applies to casting in routes.py, can we replace all casting with just directly creating the DeveloperPath with a string?